libbeat/common/mapstr_pointer.go

```diff
@@ -0,0 +1,29 @@
+package common
+
+import (
+	"sync/atomic"
+	"unsafe"
+)
+
+// MapStrPointer stores a pointer to atomically get/set a MapStr object
+// This should give faster access for use cases with lots of reads and a few
+// changes.
+// It's imortant to note that modifying the map is not thread safe, only fully
+// replacing it.
+type MapStrPointer struct {
+	p unsafe.Pointer
+}
```

Comment:
How about changing the type to:

```
type VolatileMapStr struct {
  p *unsafe.Pointer
}

func MakeVolatileMapStr(m MapStr) VolatileMapStr {
  return VolatileMapStr{ p: &unsafe.Pointer(&m) }
}

func (m VolatileMapStr) Get() MapStr {
  return *((*MapStr)(atomic.LoadPointer(m.p)))
}

func (m VolatileMapStr) Set(new MapStr) {
  atomic.StorePointer(m.p, unsafe.Pointer(&new))
}

```

Then you can use `VolatileMapStr` and `MapStr` by value, just like `MapStr` is already passed by value.

Reply:
LGTM, but don't really like the Volatile name, somehow it hides the real thing?